docs/spec/openapi.yml

```diff
@@ -72,6 +72,85 @@ paths:
           application/json:
             schema:
               $ref: '#/components/schemas/UserCredentials'
+  '/permissions':
+    get:
+      tags:
+        - Authentication & Authorization
+      summary: Get user permissions
+      description: |
+        This API returns a list of permissions for the user that is currently logged in.
+        
+        *Example:*
+        Assume the following RBAC policy, and users `alice` and `bob:
+        ```
+        p, adminrole:role, namespaces, read, *
+        p, adminrole:role, database-engines, *, */*
+        p, adminrole:role, database-clusters, *, */*
+        p, bob, database-clusters, *, */*
+
+        g, alice, adminrole:role
+        ```
+
+        The API will return the following permissions for `alice`:
+        ```
+        {
+          "permissions": [
+            [
+                "adminrole:role",
+                "namespaces",
+                "read",
+                "*"
+            ],
+            [
+                "adminrole:role",
+                "database-engines",
+                "*",
+                "*/*"
+            ],
+            [
+                "adminrole:role",
+                "database-clusters",
+                "*",
+                "*/*"
+            ]
+          ]
+        }
```

Comment:
Should we hide the info about roles (e.g. replace `adminrole:role` with `alice`)? Can there be any security implication of letting the user know about which roles are assigned to him?

Reply:
I don't think there is but just sparking some discussion :sunglasses:

Reply:
I don't think there should be any security implications. I'm actually thinking quite the opposite - since most users may not have access to the RBAC policy, this API would be the way to let them know which roles they've been assigned

But if it makes more sense, I'm fine with resolving the role names to the user..

Reply:
Yeah, I think this is fine :+1:

Reply:
Update: after speaking with @dianabirs , it was decided to resolve the roles to the username, since it would make the FE implementation easier